presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -915,6 +917,24 @@ public static double greatCircleDistance(
         return atan2(sqrt(t1 * t1 + t2 * t2), t3) * EARTH_RADIUS_KM;
     }
 
+    @SqlNullable
+    @ScalarFunction
+    @Description("Calculates the point on the first geometry that is closest to the second geometry")
```

Comment:
@jwass In a general case, there are multiple pairs of points that are located at a minimum distance. Consider the following cases:

* a circle and its center point;
* an octagon and its center point;
* two parallel lines;
* two concentric squares;
* etc.

Let's start with the use case. How are you thinking of using this function?

Reply:
Hi @mbasmanova - one of the ways I planned to use this was to very approximately compute the closest point between two geometries - say from a point to a polygon - and then use a different distance function, `great_circle_distance()` to compute the distance over the sphere. Of course there are many issues and that the closest point in (lat/lon) would not be the closest point on the sphere, but it would at least be a reasonable guess without having to implement a full geographic closest point function.